app/controllers/FileController.scala

```diff
@@ -0,0 +1,120 @@
+import ch.datascience.service.security.ProfileFilterAction
+import ch.datascience.service.utils.persistence.graph.{ GraphExecutionContextProvider, JanusGraphTraversalSourceProvider }
+import ch.datascience.service.utils.persistence.reader.VertexReader
+import ch.datascience.service.utils.{ ControllerWithBodyParseJson, ControllerWithGraphTraversal }
+import models.RenameRequest
+import play.api.Logger
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json._
+import play.api.libs.ws.WSClient
+import play.api.mvc.Controller
+
+import scala.concurrent.Future
+
+/**
+ * Created by jeberle on 25.04.17.
+ */
+@Singleton
+class FileController @Inject() (
+    config:                                         play.api.Configuration,
+    jwtVerifier:                                    JWTVerifierProvider,
+    rmJwtVerifier:                                  ResourcesManagerJWTVerifierProvider,
+    graphMutationClientProvider:                    GraphMutationClientProvider,
+    implicit val wsclient:                          WSClient,
+    implicit val graphExecutionContextProvider:     GraphExecutionContextProvider,
+    implicit val janusGraphTraversalSourceProvider: JanusGraphTraversalSourceProvider,
+    implicit val vertexReader:                      VertexReader
+) extends Controller with ControllerWithBodyParseJson with ControllerWithGraphTraversal with RequestHelper {
+
+  lazy val gc: GraphMutationClient = graphMutationClientProvider.get
+
+  lazy val logger: Logger = Logger( "application.AuthorizeController" )
+
+  def get_property( persistedVertex: PersistedVertex, name: String ) =
+    persistedVertex.properties.get( NamespaceAndName( name ) ).flatMap( v => v.values.headOption.map( value => value.asInstanceOf[StringValue].self ) )
+
+  def get_creation_time( persistedVertex: PersistedVertex ) =
+    persistedVertex.properties.get( NamespaceAndName( "system:creation_time" ) ).flatMap( v => v.values.headOption.map( value => value.asInstanceOf[LongValue].self ) )
+
+  def objectRename( fileId: Long ) = ProfileFilterAction( jwtVerifier.get ).async( bodyParseJson[RenameRequest]( RenameRequest.format ) ) { implicit request =>
+    logger.info( s"objectRename - ${request.body} - ${request.token.getSubject}" )
+
+    /* Steps:
+ *   1. Resolve graph entities
+ *   2. Request access authorization from Resource Manager
+ *   3. Validate response from RM
+ *   4. Log to Knowledge Graph
+ */
+
+    // Step 1: Resolve graph entities
+    val token: String = request.headers.get( "Authorization" ).getOrElse( "" )
+    val rmc = new ResourceManagerClient( config )
+    val g = graphTraversalSource
+    val t = g.V( Long.box( fileId ) ).out( "resource:has_location" )
+    graphExecutionContext.execute {
+      if ( t.hasNext ) {
+        vertexReader.read( t.next() ).flatMap { data =>
+          val extra = Some( Json.toJson( Map(
+            "old_name" -> get_property( data, "resource:path" ),
```

Comment:
Actually the path can be controlled by the storage backend. (We can imagine that it puts a uuid as local filename or row ID in a database with a blob column). The filename that is exposed to the user is the resource:filename.
However, it is good to inform the backend that a change has been done in case it want to propagate it too (thinking here about the local-fs backend).

Reply:
I thought about it too, and concluded that it would generate too much backend specific cases. The only thing to keep in mind is that backend path and filename can diverge.